src/System.Text.JsonLab/System/Text/Json/JsonReaderOptions.cs

```diff
@@ -4,11 +4,20 @@
 
 namespace System.Text.JsonLab
 {
-    [Flags]
-    public enum JsonReaderOptions
+    public struct JsonReaderOptions
     {
-        Default = 0b0000,       // Don't allow comments, treat as invalid json if found
-        AllowComments = 0b0001, // Allow comments but don't skip them
-        SkipComments = 0b0011,  // Allow and Skip comments
+        public JsonReaderOptions(CommentHandling commentHandling)
+        {
+            _commentHandling = commentHandling;
+        }
+
+        internal CommentHandling _commentHandling;
+
+        public enum CommentHandling
```

Comment:
Why do we want this to be backed by int and not byte?

Reply:
It can be backed by a byte. Is there a downside?

Reply:
If you use byte, the downside is that you won't be able to have more than 256 options; the upside that JsonReaderOptions might be smaller.

Reply:
We recently changed `JsonTokenType` from a byte-backed enum to be a regular int. In your opinion, should it remain byte-backed as well?

Reply:
I will make this change in an upcoming PR. Thanks!